### PR TITLE
Minor code and documentation cleanup. Closes #4618

### LINKED ISSUE #2260: README should mention that source download refers to MDK and link to the better documentation
A.) There should be a link to this page: http://mcforge.readthedocs.org/en/latest/gettingstarted/

B.) The instructions here should reference the MDK download, as that acronym is not as intuitive as you'd think an easily confused with MD5.

### LINKED ISSUE #3287: [1.10.2] Outdated doc comment on FMLPreInitializationEvent#getModLog
The doc comment on `FMLPreInitializationEvent#getModLog` says that logging can be configured through the **config/logging.properties** file, but this is no longer the case since Minecraft switched to log4j logging.

### LINKED ISSUE #3533: Update readme.md
The current readme states that the current branch is 1.10.x this should be updated to 1.11

### LINKED ISSUE #3556: [1.11] Update IShearable
`onSheared` should be `@Nonnull` and ought to return a `NonNullList<ItemStack>`. The javadocs need updating too.

Making this an issue, not a PR, as this has compatibility implications, and I'd like to know peoples' thoughts.

### LINKED ISSUE #3707: IShearable javadoc ItemStack arg
The doc in IShearable:

> @param item The itemstack that is being used, Possible to be null

Except that argument is annotated with Nonnull, as you'd except from the usage of ItemStack through the game.
Would recommend "empty" or ItemStack.EMPTY, for replacement.

Also at onSheared:

> @return A ArrayList containing all items from this shearing. Possible to be null.
Null case isn't handled by ItemShears, so it shouldn't be recommended.

### LINKED ISSUE #4618: Typos
Use this issue to list all typos in documentation and anywhere else in the codebase.
I can fix them when there are a lot listed here.

### LINKED ISSUE #4663: genPatches on a clean codebase creates modified patches
When I execute 'genPatches' on a **clean** instance of forge (freshly cloned, no file modifications whatsoever) 2 patch files are modified. I've created a commit [here](https://github.com/HoldYourWaffle/MinecraftForge-1.12-RecipeConvertingAssistance/commit/871ee4b211cf4f7e65c07bd818dd0c5aa70f2806) that contains these diffs.

Is there something wrong with my workspace? It seems like no one else has this.

### PR DESCRIPTION
**Pull request by mezz**
Fixes #2260
Fixes #3287
Fixes #3518
Fixes #3533
Fixes #3556
Fixes #3707
Fixes #4618 
Fixes #4663

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,63 +1,38 @@
...
- the 1.10.x branch for Minecraft 1.10.2.
-
-If you intend to modify a minecraft patch, or add a new one, always run
-```gradlew genPatches``` to generate your patch and validate that you
-are not doing anything unexpected in your patch.
+[See the "Making Changes and Pull Requests" section in the Forge documentation](http://mcforge.readthedocs.io/en/latest/forgedev/#making-changes-and-pull-requests).
 
 ### Contributor License Agreement
-As stated in the (https://github.com/MinecraftForge/MinecraftForge/blob/1.10.x/LICENSE-new.txt)
+As stated in the [LICENSE-new](https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/LICENSE-new.txt)
```
Comment (bs2609) [inline README.md]: Use a relative link here, and then it should work across branches in future.

Comment (mezz) [reply README.md]: Cool, I didn't know about relative links on Github.

**Review by liach — APPROVED** [review]

**Comment by covers1624** [comment]
Streams usage in ModeLoader.  
https://github.com/MinecraftForge/MinecraftForge/blob/10b8d47b538abe71eec374463dd9d94b9bf79364/src/main/java/net/minecraftforge/client/model/ModelLoader.java#L211-L212  

```java
        List<Block> blocks = StreamSupport.stream(Block.REGISTRY.spliterator(), false)
		        .filter(block -> block.getRegistryName() != null)
		        .sorted(Comparator.comparing(b -> b.getRegistryName().toString()))
		        .collect(Collectors.toList());
```  
https://github.com/MinecraftForge/MinecraftForge/blob/10b8d47b538abe71eec374463dd9d94b9bf79364/src/main/java/net/minecraftforge/client/model/ModelLoader.java#L283-L284  

```java
        List<Item> items = StreamSupport.stream(Item.REGISTRY.spliterator(), false)
		        .filter(item -> item.getRegistryName() != null)
		        .sorted(Comparator.comparing(i -> i.getRegistryName().toString()))
		        .collect(Collectors.toList());
```  
Eclipse stub comments
https://github.com/MinecraftForge/MinecraftForge/blob/10b8d47b538abe71eec374463dd9d94b9bf79364/src/main/java/net/minecraftforge/client/model/pipeline/ForgeBlockModelRenderer.java#L62-L67
https://github.com/MinecraftForge/MinecraftForge/blob/10b8d47b538abe71eec374463dd9d94b9bf79364/src/test/java/net/minecraftforge/debug/ModelAnimationDebug.java#L505-L508
Method that was never a thing, probably breaking change, but here anyway.
https://github.com/MinecraftForge/MinecraftForge/blob/10b8d47b538abe71eec374463dd9d94b9bf79364/src/main/java/net/minecraftforge/client/model/pipeline/VertexLighterFlat.java#L271

**Comment by bs2609** [comment]
Can see this for `ForgeBlockModelRenderer` cleanup:
https://github.com/bs2609/MinecraftForge/blob/4afcbf626ed111c0214cc62321c38769d1bc0953/src/main/java/net/minecraftforge/client/model/pipeline/ForgeBlockModelRenderer.java#L38-L50

**Comment by liach** [comment]
https://github.com/MinecraftForge/MinecraftForge/blob/10b8d47b538abe71eec374463dd9d94b9bf79364/src/main/java/net/minecraftforge/common/util/EnumHelper.java#L378-L382

Can be simplified to:
(import `org.apache.commons.lang3.ArrayUtils`)
```java
 T[] previousValues = (T[])valuesField.get(enumType); 
 T newValue = makeEnum(enumType, enumName, previousValues.length, paramTypes, paramValues); 
 setFailsafeFieldValue(valuesField, null, ArrayUtils.add(previousValues, newValue)); 
```
The simplification speeds up by removing unnecessary collection creations.
It would not have merge conflicts with #4656

File: mdk/README.txt [hunk 2]
```diff
@@ -33,14 +36,14 @@ If you preffer to use IntelliJ, steps are a little different.
...
-If at any point you are missing libraries in your IDE, or you've run into problems you can run "gradlew --refresh-dependencies" to refresh the local cache. "gradlew clean" to reset everything {this does not effect your code} and then start the processs again.
+If at any point you are missing libraries in your IDE, or you've run into problems you can run "gradlew --refresh-dependencies" to refresh the local cache. "gradlew clean" to reset everything {this does not affect your code} and then start the processs again.
 
 Should it still not work, 
 Refer to #ForgeGradle on EsperNet for more information about the gradle environment.
 
 Tip:
 If you do not care about seeing Minecraft's source code you can replace "setupDecompWorkspace" with one of the following:
-"setupDevWorkspace": Will patch, deobfusicated, and gather required assets to run minecraft, but will not generated human readable source code.
+"setupDevWorkspace": Will patch, deobfuscated, and gather required assets to run minecraft, but will not generated human readable source code.
```
Comment (Skgland) [inline mdk/README.txt]: Shouldn't it be deobuscat and generat rather than deobfuscated an generated?

Comment (liach) [reply mdk/README.txt]: `deobfuscate`. Use a verb.

**Comment by sokratis12GR** [comment]
https://github.com/MinecraftForge/MinecraftForge/blob/16bfd8cef1d12ee9ca0de1122addaf9916767ae9/src/main/java/net/minecraftforge/fml/relauncher/SideOnly.java#L47 the `public` could be removed from the SideOnly annotation
Same goes for WorldAccessContainer 
https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/src/main/java/net/minecraftforge/fml/common/WorldAccessContainer.java#L30L31

**Comment by CDAGaming** [comment]
Going to Add one more Reference for this:
Closes #4582 (Though I'll probably just redo it later for doLast should the day come)

**Comment by covers1624** [comment]
`IRenderContextHandler` is a dead class, wasn't even used in 1.7, can we add a TODO for 1.13 removal?

**Comment by sokratis12GR** [comment]
Hmm, I this a good place to suggest the depreciation of `ITileEntityProvider`?

**Comment by mezz** [comment]
`ITileEntityProvider` is a minecraft class, why deprecate it?

**Comment by sokratis12GR** [comment]
Modders shouldn't really use it, Its not needed outside vanilla, because the Block class already provides hasTileEntity and createTileEntity methods. A lot of beginners and quite a bit of older codes contain the implementation of ITileEntittProvider when it really shouldn't be used anymore

**Comment by mezz** [comment]
I think it should be considered in a different PR.

**Comment by mezz** [comment]
This is looking pretty good right now so I'm going to merge it before it gets too out of hand :)

**Comment by sokratis12GR** [comment]
Alrighty, I know it by itself isn't worth a PR tho, so /shrug :)  This one so far looks awesome,  great job :D

**Pull request merged**